contracts/FeralfileEnglishAuction.sol

```diff
@@ -0,0 +1,378 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "./IFeralfileSaleData.sol";
+import "./ECDSASigner.sol";
+import "./IFeralfileVault.sol";
+
+contract IFeralfileExhibitionV4 is IFeralfileSaleData {
+    function buyArtworks(
+        bytes32 r_,
+        bytes32 s_,
+        uint8 v_,
+        SaleData calldata saleData_
+    ) external payable {}
+}
+
+contract FeralfileEnglishAuction is Ownable, IFeralfileSaleData, ECDSASigner {
+    struct EnglishAuction {
+        bytes32 id;
+        uint256 startAt;
+        uint256 endAt;
+        uint256 extendDuration;
+        uint256 extendThreshold;
+        uint256 minIncreaseFactor;
+        uint256 minIncreaseAmount;
+        uint256 minPrice;
+        bool isSettled;
+    }
+
+    struct Bid {
+        address bidder;
+        uint256 amount;
+        bool fromFeralFile;
+    }
+
+    struct LatestAuctionStatus {
+        Bid latestBid;
+        uint256 endAt;
+        bool isSettled;
+    }
+
+    mapping(bytes32 => EnglishAuction) public auctions;
+    mapping(bytes32 => Bid) public auctionLatestBid;
+
+    constructor(address signer_) ECDSASigner(signer_) {}
+
+    function getAuctionID(
```

Comment:
I believe we don't need to maintain a specific logic for the auctionID, we just need to create rule on the backend, similar to what we did for token mint. 
The reason i think is 
- Create more complexity to maintain a specific logic. We also need to think about the compatibility if we update the logic in the future. 
- Add more computation for every uses. 

Suggestion: 
Use specific algorithm like UUID->decimal to maintain the auctionID. It could be even more easier on backend side.